### PR TITLE
completing view single article

### PR DESCRIPTION
**Pull request by althrntn**

### CODE REVIEW

File: src/components/SingleArticle.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { getArticlebyId } from "../utils/api_funcs"
+
+const SingleArticle = () => {
+    const [selectedArticle, setSelectedArticle] = useState({})
+    const {article_id} = useParams()
+     const [isLoading, setIsLoading] = useState(true)
+ 
+    useEffect(()=> {
+        getArticlebyId(article_id).then((response) => {setSelectedArticle(response.article); setIsLoading(false)})
+    })
```
Comment (popatre) [inline src/components/SingleArticle.jsx]: This is missing the dependency array - its currently constantly doing this request

Comment (popatre) [reply src/components/SingleArticle.jsx]: Its also just missing some error handling for if I got to an id that isnt there. 
At the moment, it just shows a constant loading state

**Review by popatre — CHANGES_REQUESTED** [review]

File: src/components/ArticleCard.jsx [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
+
 const ArticleCard = ({article}) => {
     const date = article.created_at
+    const navigate = useNavigate()
     return(<li key={article.article_id} className='article_card'> <h1>{article.title}</h1>
-    <p> By {article.author}, {date.slice(8, 10)}/{date.slice(5, 7)}/{date.slice(0, 4)}</p>
+    <p> By {article.author}, {date.slice(8, 10)}-{date.slice(5, 7)}-{date.slice(0, 4)}</p>
     <p>Topic: {article.topic}</p>
-    <button className="article_button">Read in full</button></li>)
+    <button className="article_button" onClick={(e)=>{navigate(`/articles/${article.article_id}`)}}>Read in full</button></li>)
```
Comment (HMelesi) [inline src/components/ArticleCard.jsx]: You don't need the e event parameter here as you don't use it, I'd just leave it empty.

**Review by HMelesi — APPROVED** [review]
Looks good Alice!

I was just having a look around your app generally and you're getting some warnings about keys in the console. Just needs fixing at some point. Also some weird stuff happens when you resize the screen! Styling comes on a later ticket so don't worry about it now, but just wanted to point out while I noticed :)

**Pull request merged**